stagemonitor-core/src/main/java/org/stagemonitor/core/CorePlugin.java

```diff
@@ -396,6 +396,44 @@
 			.configurationCategory(CORE_PLUGIN_NAME)
 			.tags("elasticsearch", "advanced")
 			.buildWithDefault(5);
+	private final ConfigurationOption<Boolean> enableSpringCloudConfigConfigurationSource = ConfigurationOption.booleanOption()
+		.key("stagemonitor.configuration.springcloud.enabled")
+		.dynamic(false)
+		.label("Enable or disable the Spring Cloud Config configuration source. Default is false")
+		.description("Use to let stagemonitor pull configuration from a Spring Cloud Config server via http. It has the lowest precedence." +
+			"{stagemonitor.applicationName} will be used as the application name to fetch configuration for." +
+			"If the config server can't be reached and the flag is true stagemonitor will be disabled.")
+		.configurationCategory(CORE_PLUGIN_NAME)
+		.buildWithDefault(false);
+	private final ConfigurationOption<String> springCloudConfigServerAddress = ConfigurationOption.stringOption()
+		.key("stagemonitor.configuration.springcloud.address")
+		.dynamic(false)
+		.label("Address of the Spring Cloud Config server")
+		.description("Must be a http or https url.")
+		.configurationCategory(CORE_PLUGIN_NAME)
+		.buildWithDefault("");
+	private final ConfigurationOption<Collection<String>> springCloudConfigConfigurationSourceProfiles = ConfigurationOption.stringsOption()
```

Comment:
Maybe instead of adding the profiles like this, just use full urls for `stagemonitor.configuration.springcloud.address`. For example `http://config.example.com/petclinic-common.properties,http://config.example.com/petclinic-prod.properties`. That way the configuration source is not specific to spring cloud config but can be reused for any remote properties file.

Reply:
I like the idea and will change the code accordingly.